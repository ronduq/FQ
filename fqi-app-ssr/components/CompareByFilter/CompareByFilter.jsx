import React from 'react';
import classNames from 'classnames';
import Icon from '../Icon/Icon';
import RadioButton from '../../components/RadioButton/RadioButton';
import t from '../Icon/Icon.module.scss';
import s from './CompareByFilter.module.scss';
import get from 'lodash.get';


const CompareByFilter = ({sortRanks, selected, showPerception, content}) => {
  const radioGroup="compareBy"
  return (
    <fieldset className={s.container}>
      <legend>{get(content,'compare')}</legend>
      <RadioButton radioGroup={radioGroup} label={get(content,'quality')} id='quality' changeAction={sortRanks} selected={selected === 'quality'}/>
      <RadioButton radioGroup={radioGroup} label={get(content,'value')} id='value' changeAction={sortRanks} selected={selected === 'value'}/>
      { showPerception && 
        <RadioButton radioGroup={radioGroup} label={get(content,'perception')} id='perception' changeAction={sortRanks} selected={selected === 'perception'}/> }
    </fieldset>
  );
};

CompareByFilter.defaultProps = {
  showPerception: true
};

CompareByFilter.propTypes = {
};

export default CompareByFilter;
