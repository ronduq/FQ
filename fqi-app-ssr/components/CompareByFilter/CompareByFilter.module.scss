
@import '../../style/settings/colours';

.container {
  background: white;
  padding: 45px 15px 15px;
  border-left: 1px solid $colour-grey-medium;
  border-right: 1px solid $colour-grey-medium;
  border-bottom: 1px solid $colour-grey-medium;
  position: relative;

  legend {
    position: absolute;
    top: 10px;
    left: 25px;
  }
}