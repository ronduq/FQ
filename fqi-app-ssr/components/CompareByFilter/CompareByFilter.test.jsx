/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import CompareByFilter from './CompareByFilter'

describe('With Snapshot Testing', () => {
  const props = {
    selected: 'quality',
    sortRanks: () => {},
  }

  it('CompareByFilter snapshot"', () => {
    const component = renderer.create(<CompareByFilter { ...props } />)
    const componentJSON = component.toJSON()
    expect(componentJSON).toMatchSnapshot()
  });

  it('should show perception radio input when showPerception is true', () => {
    const wrapper = mount(<CompareByFilter {...props} showPerception/>);
    expect(wrapper.find('input').length).toBe(3)
  });

  it('should not show perception radio input when showPerception is true', () => {
    const wrapper = mount(<CompareByFilter {...props} showPerception={false} />);
    expect(wrapper.find('input').length).toBe(2)
  });
})
