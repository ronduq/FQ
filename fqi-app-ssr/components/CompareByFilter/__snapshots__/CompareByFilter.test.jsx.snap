// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing CompareByFilter snapshot" 1`] = `
<fieldset
  className="container"
>
  <legend />
  <div
    className="radio"
  >
    <input
      autoComplete="off"
      checked={true}
      id="quality"
      name="compareBy"
      onChange={[Function]}
      tabIndex="0"
      type="radio"
      value="quality"
    />
    <label
      htmlFor="quality"
    />
  </div>
  <div
    className="radio"
  >
    <input
      autoComplete="off"
      checked={false}
      id="value"
      name="compareBy"
      onChange={[Function]}
      tabIndex="0"
      type="radio"
      value="value"
    />
    <label
      htmlFor="value"
    />
  </div>
  <div
    className="radio"
  >
    <input
      autoComplete="off"
      checked={false}
      id="perception"
      name="compareBy"
      onChange={[Function]}
      tabIndex="0"
      type="radio"
      value="perception"
    />
    <label
      htmlFor="perception"
    />
  </div>
</fieldset>
`;
